v2/tools/generator/internal/codegen/pipeline/add_arm_conversion_interface.go

```diff
@@ -231,20 +237,13 @@ func (c *armConversionApplier) addARMConversionInterface(
 		return emptyDef, errors.Errorf("ARM definition %q did not define an object type", armDef.Name())
 	}
 
-	// Determine if we need special handling for collection properties. Some RPs we need to send empty collections rather
-	// than nil collections, we need to determine if this def is subject to this requirement
-	payloadType := config.OmitEmptyProperties
-	if pt, ok := c.config.PayloadType.Lookup(kubeDef.Name().InternalPackageReference()); ok {
-		payloadType = pt
-	}
-
 	addInterfaceHandler := func(t *astmodel.ObjectType) (astmodel.Type, error) {
 		result := t.WithInterface(armconversion.NewARMConversionImplementation(
 			armDef.Name(),
 			objectType,
+			kubeDef.Name(),
 			c.idFactory,
-			typeType,
-			payloadType))
+			typeType))
```

Comment:
Should we consider renaming `typeType` to `typeKind`?

Reply:
Good catch - done